### ISSUE TITLE
Archery "max recoil fix" recent commit breaks archery.

### ISSUE THREAD

**Issue by Amariithynar** [CleverRaven/Cataclysm-DDA#24411]
Not sure why #24390 was made then immediately merged in without any discussion, so opening an issue regarding it as was recommended by @Firestorm01X2, the creator of that commit, copying over my complaint and sourcing from that discussion.

>This was a badly implimented change. Once you've drawn a bead with a bow, it takes minimal effort to fire another arrow to land at the same general location. Most of the time spent is in drawing another arrow and nocking it to the bow, and that doesn't take much time at all. You don't hold the bow at full draw for long periods of time, either, which is what's actually humanly impossible (since we just don't have the strength to do so for the draw weights we use; mutants or cyborgs might, however). According to http://cddawiki.chezzo.com/cdda_wiki/index.php?title=Turn a turn is six seconds, and by default takes 100 move points. While untrained archers certainly wouldn't be as skilled at the process (really, it's training your body to go through the motions without thinking about them) it's not hard to fire arrows at much faster speeds accurate to the previous timeframes at full steadiness, or faster; https://www.youtube.com/watch?v=Lnv9Pr3AWSw and I know I'm likely to get flack for referencing him, but https://www.youtube.com/watch?v=BEG-ly9tQGk (honestly, Lars should be considered to be like, Archery 10 considering his high skill with the bow, but it gets the point across). Accuracy is not negatively impacted by rapidity of shot, either, which IS accurate (https://www.youtube.com/watch?v=1o9RGnujlkI).
>
>There needs to either be a scalar according to archery skill that reduces the recoil effect until at an average rank (4? 5?) it's completely reverted to how it was before, the effect needs to be drastically reduced and bow speeds need to be increased anyways (it shouldn't take 18 seconds to aim and accurately fire a bow, period, even for an amateur; bow damage should drop, though), or this simply needs to be reverted wholesale.

Firestorm replied at this point with,
>Well, I partally agree. But it has to be fixed since before you don't need to retarget at all. Also skill already affects aiming speed so now it will be taken in effect since now you need to target after each shot. Also don't forget that slings use the same mechanics.

to which I replied with,
>Higher skill does increase overall speed, but I'm talking specifically the negative effect of this change needing to be scaled according to skill. This is basically a double penalty to archery where you are acting like they need to put the bow down between every shot and need to hold the arrow to their cheek for an inordinate amount of time for each shot if they want to be even somewhat accurate. Archery 0 character, it makes sense; I've seen a lot of people who've never picked up a bow at a few archery ranges that do take that long, trying to just get used to putting everything together; they're just trying to learn the absolute basics, after all. But at Archery 10, where 10 is supposed to be "amongst the top in the world"? Using a composite bow to take a shot now takes 194 move points without any bow mods. That's nearly two turns, or about 12 seconds. And it doesn't reach that until Archery 8, either. Archery 6 now takes 244 points, or about 15 seconds. Even for "regular aim", Archery 6 now is 185 points or about 11 seconds, and Archery 8 now is 135 or about 8 seconds; twice that of a competent archer. Unlike with a rifle, where recoil forces you to setup your shot every single time, there really isn't recoil with bows. You can shoot and shoot and shoot, accurate and fast, to multiple targets, until you run out of arrows, unless you need to swing your aim wildly away from the general fire arc that you had it at before; but that was already in the game! Changing your firing arc lost steadiness. Furthermore, the whole reason that rifles and crossbows need to re-aim between reloads is because you have to stop looking down the sights and bring the weapon down from ready to swap magazines, as you can't afford to just discard them with a quick-release and hotswap like soldiers have trained long and hard to be able to do (https://www.youtube.com/watch?v=W-6jXQ3FE4o), or set the next bolt in the groove.
>
>The only thing I'd agree with needing a fix is the sling, because of the long windup time between each shot, and since you're releasing one end of the sling to let the bullet loose, requiring you to set back up completely after every shot. That, to me, says that sling needs to not be considered part of archery, not that bows needed to be messed with, especially not when they don't have recoil (well, they don't so long as you're not trying to pull a draw weight beyond your strength, anyways).
>
>A far better solution to making bows less desireable over other weapons would have been to increase the stamina cost of firing a bow. Even shots with a 20 lb. bow takes a lot more energy out of you than raising a crossbow or rifle does. The primary reason to use a bow is not high damage, but how relatively quiet it is to fire, creating minimal noise until the arrow thunks into the target.

**Comment by Lasiace** [comment]
Having tried out a bit of recreational archery as a complete amateur a little while back, I can say that drawing a bow is _tiring._ As a result of that, my instructor advised me to quickly draw the bow and release just as quickly, without worrying myself over aiming. In fact, draw it too long, and your aim will suffer instead, due to your arms shaking from the strain, so in that sense it's the opposite of aiming with a gun, your aim starts out reasonable and gets worse the longer you aim.

Basically what I'm saying is this change is kind of inaccurate in making bows take so long to aim. You **can’t** aim for a long time, not unless you want your aim to get _worse,_ that’s how bows work. And precisely because you can’t aim for a long time, your shots won’t be as accurate until you have the muscle memory to aim well from the get-go. So I think how bows _should_ work is that they actually reach "precise" aim very quickly, but that “precise” aim won’t actually be very accurate until higher skill levels, where you will have the knowledge and experience to know where to aim everything in that short span of time you have to do so.

As for having to re-aim every time – a little bit sure, but certainly not resetting each time. The arm holding the bow actually doesn’t need to move much, it's more just a guide, the arm firing the arrow is the one that does all the work, so you won’t really need to re-aim much in that case.

That's my take on it at least. Do note, I’m not an expert, but based on what I do know and have learned and tried of archery, this seems like the wrong direction to take bows. It doesn't seem accurate at all to me. 

As for making bows take more stamina, well I can definitely say my arms were killing me after those archery sets, that’s for sure. Might just be because I’m a weakling though, haha

TL;DR: You shouldn't aim a bow much in the first place since your arms will get tired and screw with your aim anyway, so making aiming take longer to become precise is the wrong way of doing things, rather making bows less accurate (until higher skill maybe) but faster seems more in-line with reality to me.

**Comment by Amariithynar** [comment]
People who I know hit the gym all the time and are really good at both endurance sports like long distance running and in lifting weights, get tired and sore arms fairly quickly when they have to shoot a whole bunch at once. Every time I've gone with them to do some archery at a range, I've always had to tap out early because of just how sore my arms get, and while I'm not in the best of shape, I am a cook, used to using my arms and back all day, lifting heavy loads and moving them around, and I have made my own chainmail and gone to some local HEMA and similar events.

**Comment by Firestorm01X2** [comment]
Thete is problem - bows now are guns with no recoil and witj automated reloading Qol mode.

Implementing new inique aming mode for bows is hard.

There is easier way - reset recoil base on archery skill.

**Comment by Amariithynar** [comment]
Steadiness being set to base 50% would help over what it is now, but it still would be taking far too long to aim. An average archer can fire an arrow, accurately, about every 4 seconds once they've found their mark (the initial aiming period to get a steady shot). That's about 70 move units (4.2 seconds). At archery 7, with 15 strength, it now takes 266 move points for me to fire accurately (precisely), every shot. That's about 16 seconds per shot. to just REGULARLY aim and fire, 206 move units. This is also entirely ignoring hand encumbrance. This is just rediculous, and not reasonable by any measure. The M1 Garand hits four times harder (76 damage~ per shot vs. 25~30 damage per shot) and *has a faster fire rate* at rifles 6, taking 268 for the first shot, and then only 209 for later accurate shots, until the clip empties and you need to swap the clip, or have to move.

**Comment by Amariithynar** [comment]
The number I quoted is directly backed by multiple video evidence linked above, and easily found elsewhere. Lars Andersen can fire much faster than that. That said, rifles and even crossbows do take too long to re-aim as well, even with high-caliber, high-recoil ammunition, when you're only shooting semi-auto or bolt-action, but that's beyond this issue. Again. 4 shots at 15 seconds per shot means a FULL MINUTE to take those shots. It does NOT take that long to fire a bow, for even the most amateur archers. Archery WILL tire you out much faster, because you're using physical energy, not chemical, to propel your shots. Crossbows are the rare hybrid where it actually makes some sense to reset every shot, since you can't reset the string onto the catch and reseat a bolt, all while maintaining your aim- but they shouldn't take forever to pull up and fire, either.

**Comment by Firestorm01X2** [comment]
Does someone remember how bows used to behave before implementation of RELOAD_AND_SHOOT mechanics?

They supposed to be resetted after reload. Right?

**Comment by Amariithynar** [comment]
Bows did not retarget after reload prior to RELOAD_AND_SHOOT. They only required retargeting if you drastically changed your firing arc (something like 90 degrees reduced your steadiness by like 50%).

**Comment by Amariithynar** [comment]
I can't say if it was intended behaviour or not- Would have to ask @kevingranade that. However, it is *accurate* behaviour. If anything, other ranged weapons need to have their moves per attack reduced even further.

**Comment by darktoes** [comment]
Personally I'm more or less in favour of the recoil mechanic. While the bow doesn't actually move very much after firing, assuming it's tuned properly, the archer still has to reach down/back, get another arrow, nock the thing, and draw again. If you can hold your offhand out at full length while doing all that without moving it an inch then you must be some sort of robot.
Your entire body is involved in drawing an arrow, saying that the hand holding the bow won't move is ridiculous.

That said, I agree that aiming bows should be much faster. At short range, actual aiming time after drawing would likely be a second or two, but at long range it probably wouldn't exceed more than 3-5 seconds. With a compound you can aim for a much longer period of time, but you probably aren't going to get that much out of it. Considering the size of a bow, lining it up is surprisingly easy. An amateur isn't going to hit much at any range though. Unless they have a tuned sight on the bow.

**Comment by Amariithynar** [comment]
I actually sourced several videos where archers effectively do just that, hah. The video I linked actually shows him firing the way he normally does and hitting a dummy wearing chainmail and a gambeson and piercing through them both, but yeah, overall everyone I know in archery sees a lot of what he does as basically just being trickshotting. Also, I'd argue that "long time" is still relative to his rapid fire shots. ;P

**Comment by darktoes** [comment]
And yet in every single one of those videos the archer moves the bow back and forth while drawing. Which happens. The bow isn't steady between draws, it moves a lot.

That gambeson and chainmail are probably cheap crap or something. He doesn't put enough effort in to actually put any power behind the arrow.

**Comment by Amariithynar** [comment]
I wouldn't call less than a foot of full forward and back motion along the same area "a lot". The bow in both cases also remains roughly in line with the point being aimed at, as well. It's not jerking up and away, but moving in a fluid motion in a very small range of motion.

**Comment by Amariithynar** [comment]
Not when the displacement is towards and then away from the body, resetting to the same location that one is firing from, moving less than a foot in total back and forward, which you'd know is what is being talked about if you saw the linked video in question.

And, again, I am not advocating for no change to archery, just not what WAS implemented, especially when it was done without discussion for such a major change; previously the lack of recoil effect was relatively accurate enough to how you don't need to take more time than it took to nock and draw the arrow back to fire to finish aiming after you've already placed your initial shot, or taken your time to aim in the first place, since you could get your shots down to about 1 every 4 seconds (70 move points) at archery 7 with a longbow. Now it takes over 10 seconds for the same general accuracy.

Using that code blurb and description, "archery_ability" could probably be something like strength / 2 round to nearest + dexterity / 2 round down + archery skill * 1.75 round up + marksmanship; so someone with 8 str and dex and 5 archery and 3 marksmanship would look like 4 + 4 + 9 + 3 = 20. Steadiness has 30 points to the meter, so that means an average person that's a pretty skilled hobbyist would only lose 10 steadiness per shot, or about 100 move points. At archery 8 marksmanship 5, that would go up to 27. 9 archery and 6 marksmanship would go up to the cap of 30. So at archery 9 and 6 marksmanship with default strength and dexterity, you'd be able to pull all the stunts that are shown in the video above. Even with that formula, that's still pretty bad that it takes that long, but it's definitely better than current.

**Comment by darktoes** [comment]
Thinking more on it, I still disagree that the bow hand stays in place between shots, but I don't think recoil is the way to go about it, although you could argue it's to model the bow is stuck out in firing position after a shot rather than closer to the body where it can be nocked/drawn. If recoil is used to model that it should only be a tiny amount, perhaps a quarter of a second or less.

**Comment by Firestorm01X2** [comment]
This should help:
https://github.com/CleverRaven/Cataclysm-DDA/pull/24437

**Comment by Firestorm01X2** [comment]
Should not it be closed?
